meson.build

```diff
@@ -50,6 +50,13 @@ m_dep = cc.find_library('m', required : false)
 if m_dep.found()
   add_project_link_arguments('-lm', language : 'c')
 endif
+if host_machine.system() == 'os400'
+  add_global_arguments('-pthread', language : 'cpp')
```

Comment:
We already should have
```meson
thread_dep = dependency('threads')
```
in the places where it is needed. That's the more portable way of doing this - see https://mesonbuild.com/Threads.html#threads. Is it really needed here? What happens if you leave it out?

Reply:
Yes, as I stated above, without it , it would only(maybe more, but stop ) on contorl.cc as below:
```
[1386/1566] c++ -Iscipy/optimize/_highs/libipx.a.p -Iscipy/optimize/_highs -I../scipy/optimize/_highs -I../scipy/_lib/highs/src/ipm/ipx/include -I../scipy/_lib/highs/src/ipm/basiclu/include -I../scipy/_lib/highs/src -I../scipy/_lib/highs/extern -I../scipy/optimize/_highs/cython/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++14 -O2 -g -fPIC -Wno-class-memaccess -Wno-format-truncation -Wno-non-virtual-dtor -Wno-sign-compare -Wno-switch -Wno-unused-but-set-variable -Wno-unused-variable '-DCMAKE_BUILD_TYPE="RELEASE"' -DFAST_BUILD=ON '-DHIGHS_GITHASH="n/a"' '-DHIGHS_COMPILATION_DATE="2021-07-09"' -DHIGHS_VERSION_MAJOR=1 -DHIGHS_VERSION_MINOR=2 -DHIGHS_VERSION_PATCH=0 -DHIGHS_DIR=/QOpenSys/home/zhanggan/prepackage/scipy/scipy/scipy/optimize/_highs/../../_lib/highs -UOPENMP -UEXT_PRESOLVE -USCIP_DEV -UHiGHSDEV -UOSI_FOUND -MD -MQ scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o -MF scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o.d -o scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o -c ../scipy/_lib/highs/src/ipm/ipx/src/control.cc
FAILED: scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o
c++ -Iscipy/optimize/_highs/libipx.a.p -Iscipy/optimize/_highs -I../scipy/optimize/_highs -I../scipy/_lib/highs/src/ipm/ipx/include -I../scipy/_lib/highs/src/ipm/basiclu/include -I../scipy/_lib/highs/src -I../scipy/_lib/highs/extern -I../scipy/optimize/_highs/cython/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++14 -O2 -g -fPIC -Wno-class-memaccess -Wno-format-truncation -Wno-non-virtual-dtor -Wno-sign-compare -Wno-switch -Wno-unused-but-set-variable -Wno-unused-variable '-DCMAKE_BUILD_TYPE="RELEASE"' -DFAST_BUILD=ON '-DHIGHS_GITHASH="n/a"' '-DHIGHS_COMPILATION_DATE="2021-07-09"' -DHIGHS_VERSION_MAJOR=1 -DHIGHS_VERSION_MINOR=2 -DHIGHS_VERSION_PATCH=0 -DHIGHS_DIR=/QOpenSys/home/zhanggan/prepackage/scipy/scipy/scipy/optimize/_highs/../../_lib/highs -UOPENMP -UEXT_PRESOLVE -USCIP_DEV -UHiGHSDEV -UOSI_FOUND -MD -MQ scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o -MF scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o.d -o scipy/optimize/_highs/libipx.a.p/.._..__lib_highs_src_ipm_ipx_src_control.cc.o -c ../scipy/_lib/highs/src/ipm/ipx/src/control.cc
In file included from ../scipy/_lib/highs/src/parallel/HighsSpinMutex.h:24,
                 from ../scipy/_lib/highs/src/parallel/HighsMutex.h:17,
                 from ../scipy/_lib/highs/src/parallel/HighsParallel.h:16,
                 from ../scipy/_lib/highs/src/ipm/ipx/src/control.cc:1:
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread:85:13: error: '__gthread_t' does not name a type; did you mean 'pthread_t'?
   85 |     typedef __gthread_t   native_handle_type;
      |             ^~~~~~~~~~~
      |             pthread_t
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread:90:7: error: 'native_handle_type' does not name a type
   90 |       native_handle_type _M_thread;
      |       ^~~~~~~~~~~~~~~~~~
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread:96:28: error: expected ')' before '__id'
   96 |       id(native_handle_type __id) : _M_thread(__id) { }
      |         ~                  ^~~~~
      |                            )
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread:195:5: error: 'native_handle_type' does not name a type
  195 |     native_handle_type
      |     ^~~~~~~~~~~~~~~~~~
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread: In constructor 'std::thread::id::id()':
/qopensys/pkgs/lib/gcc/powerpc-ibm-aix6.1.0.0/10/include/c++/thread:93:23: error: class 'std::thread::id' does not have any field named '_M_thread'
   93 |       id() noexcept : _M_thread() { }
      |                       ^~~~~~~~~
... ... ...
```

Reply:
Thanks, that is helpful. This is a bug in the build config that is not specific to just OS/400. In `scipy/_lib/highs` there is usage of `std::thread` without a dependency on a threading library. Let me open a separate PR for that.

Reply:
k. thanks. Then you mean I can remove the "-pthread" line from my PR?

Reply:
Yes indeed. Can you check that that works in combination with gh-17200? It's possible that there's some other code using threading, it's quite hard to figure out reliably.

Reply:
Thanks for your quick action on it. I verified it together with my PR(updated to removing the "-pthread" fix). Both "mason compile --verbose" and "python3 -m pip install ." works correctly on my system. 

Reminder , if you can , when you merge it , please mark both of the PRs as candidate for 1.9.3. Thanks.

Reply:
Great, thanks @zheddie!